runtime/stdlib/cadence_v0.42_to_v1_contract_upgrade_validator.go

```diff
@@ -152,17 +167,24 @@ func (validator *CadenceV042ToV1ContractUpdateValidator) idAndLocationOfQualifie
 	// and in 1 and 2 we don't need to do anything
 	typIdentifier := typ.Identifier.Identifier
 	rootIdentifier := validator.TypeComparator.RootDeclIdentifier.Identifier
-	location := validator.underlyingUpdateValidator.location
 
-	foundLocations := validator.TypeComparator.foundIdentifierImportLocations
+	newImportLocations := validator.TypeComparator.foundIdentifierImportLocations
+	oldImportLocations := validator.TypeComparator.expectedIdentifierImportLocations
 
-	if typIdentifier != rootIdentifier && foundLocations[typIdentifier] == nil {
-		qualifiedString = fmt.Sprintf("%s.%s", rootIdentifier, qualifiedString)
-		return common.NewTypeIDFromQualifiedName(nil, location, qualifiedString), location
+	// Here we only need to find the qualified type ID.
+	// So check in both old imports as well as in new imports.
```

Comment:
In migrations, we also allow changing types to a new type that is in a different import location. e.g: from `MetadataViews.Resolver` to `ViewResolver.Resolver`.
So when computing the type ID for the old type, it will not be found in the new import locations (i.e: `foundIdentifierImportLocations`). So we need to check in the old import locations (`expectedIdentifierImportLocations`) as well.